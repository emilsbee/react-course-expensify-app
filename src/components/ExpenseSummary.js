import React from 'react'
import selectExpenses from '../selectors/expenses'
import expensesTotal from '../selectors/expenses-total'
import { connect } from 'react-redux'
import numeral from 'numeral'


export const ExpenseSummary = ({ expenses, filters }) => {
    const visibleExpenses = selectExpenses(expenses, filters)
    const totalExpenses = expensesTotal(visibleExpenses)

    return (
        <div>
        {visibleExpenses.length === 1 ? 
        <p>Viewing {visibleExpenses.length} expense totalling {numeral(totalExpenses / 100).format('$0,0.00')}</p>
        :
        <p>Viewing {visibleExpenses.length} expenses totalling {numeral(totalExpenses / 100).format('$0,0.00')} </p>
        }
    </div>
    )
}

const mapStateToProps = (state) => {
    return {
        expenses: state.expenses,
        filters: state.filters
    }
}

export default connect(mapStateToProps)(ExpenseSummary)
