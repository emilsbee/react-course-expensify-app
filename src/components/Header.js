import { NavLink } from 'react-router-dom'
import React from 'react'; 

const Header = () => (
    <header>
        <h1>Expensify</h1>
        <p>
            <NavLink exact to="/" activeClassName="is-active">Dashboard</NavLink>
        </p>
        <p>
            <NavLink to="/create" activeClassName="is-active">Crete expense</NavLink> 
        </p>
        <p>
            <NavLink to="/help" activeClassName="is-active">Help</NavLink>
        </p>
    </header>
)

export default Header