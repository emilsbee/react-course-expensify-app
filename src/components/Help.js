import {BrowserRouter, Route, Switch, Link, NavLink} from 'react-router-dom'
import React from 'react'; 

const Help = () => (
    <div>
        This is from HelpPage 
    </div>
)


export default Help