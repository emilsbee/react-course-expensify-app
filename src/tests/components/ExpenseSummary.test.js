import React from 'react'
import { shallow } from 'enzyme'
import {ExpenseSummary}  from '../../components/ExpenseSummary'
import expenses from '../fixtures/expenses'
import {filters} from '../fixtures/filters'

test('should render expense summary with 1 expense correctly', () => {
    const wrapper = shallow(<ExpenseSummary expenses={[expenses[1]]} filters={filters}/>)
    expect(wrapper).toMatchSnapshot()
})

test('should render expense summary with 0, 2 or more expenses correctly', () => {
    const wrapper = shallow(<ExpenseSummary expenses={expenses} filters={filters}/>)
    expect(wrapper).toMatchSnapshot()
})