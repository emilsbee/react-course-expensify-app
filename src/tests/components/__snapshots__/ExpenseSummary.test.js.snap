// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense summary with 0, 2 or more expenses correctly 1`] = `
<div>
  <p>
    Viewing 
    3
     expenses totalling 
    $1,141.95
     
  </p>
</div>
`;

exports[`should render expense summary with 1 expense correctly 1`] = `
<div>
  <p>
    Viewing 
    1
     expense totalling 
    $1,095.00
  </p>
</div>
`;
